domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -631,7 +631,7 @@ public void startTx(Handler<AsyncResult<SQLConnection>> done) {
       }
       try {
         SQLConnection pgTransaction = new SQLConnection(res.result(),
-            res.result().begin(), null);
+            res.result().begin().result(), null);
```

Comment:
begin() is a Future<Transaction>. It takes some time until the database has responded to the begin request.
begin().result() will be null.
You need some async handling, for example

        res.result().begin()
        .map(transaction -> new SQLConnection(res.result(), transaction, null))
        .onComplete(done);

Reply:
done.